keyboard/src/modules/Keyboard.rb

```diff
@@ -1504,6 +1483,43 @@ def write_udev_rule
     end
   end
 
+  # Checks if the graphical environment is being executed remotely using
+  # "ssh -X"
+  def x11_over_ssh?
+    display = Builtins.getenv("DISPLAY") || ""
+    display.split(":")[1].to_i >= 10
```

Comment:
I suspect that 10 comes from the default value of X11DisplayOffset. You know, one of those things that _almost_ nobody changes.
http://becausewecan.org/node/65

But my lack of understanding on X11 over ssh and all the weird scenarios that resulted in somebody filling bnc#539218, makes me be conservative here :-)